prov/gni/configure.m4

```diff
@@ -0,0 +1,160 @@
+dnl Configury specific to the libfabrics GNI provider
+
+dnl Called to configure this provider
+dnl
+dnl Arguments:
+dnl
+dnl $1: action if configured successfully
+dnl $2: action if not configured successfully
+dnl
+
+# _PKG_CONFIG2([VARIABLE], [COMMAND1], [COMMAND2], [MODULES])
+# ---------------------------------------------
+m4_define([_PKG_CONFIG2],
+[if test -n "$$1"; then
+	pkg_cv_[]$1="$$1"
+elif test -n "$PKG_CONFIG"; then
+	PKG_CHECK_EXISTS([$4],
+		[pkg_cv_[]$1=`$PKG_CONFIG --[]$2 --[]$3 "$4" 2>/dev/null`
+		test "x$?" != "x0" && pkg_failed=yes ],
+		[pkg_failed=yes])
+else
+	pkg_failed=untried
+fi[]dnl
+])# _PKG_CONFIG2
+
+# PKG_CHECK_MODULES_STATIC(VARIABLE-PREFIX, MODULES, [ACTION-IF-FOUND],
+# [ACTION-IF-NOT-FOUND])
+#
+#
+# Variation of PGK_CHECK_MODULES which also defines $1_LIBS_STATIC
+# by using the pkg-config --libs --static pkg-config option.
+#
+# --------------------------------------------------------------
+AC_DEFUN([PKG_CHECK_MODULES_STATIC],
```

Comment:
I'd forgotten about the split I made for using ```pkg-config``` for the cray configure, so as to be able to import a unmodified pkg.m4 into the common config directory.  If you're okay with it, we can do like is in my favorite github project and put the pkg config extensions along with the content of pkg.m4 into a fi_pkg.m4 in the common config directory. 

I recall having to use ```pkg-config``` with the ```--static``` option in order to link the criterion unit tests properly.  Hence the extensions.